Supplier number

MikiDi opened the pull request:
- slim down model
- allow adding supplier number in creation form
- link out to supplier both in supplier overview and in product detail

Might still be desired:
- show supplier number literally somewhere else than in the link out
- ability to edit supplier number when editing supplier

MikiDi committed:
clarify meaning of nummer

The number next to supplier in a product detail context is a product number

MikiDi committed:
slim down business entity model, add identifier

MikiDi committed:
add form entry for supplier identifier

MikiDi committed:
add links out to supplier

in supplier overview and prod detail view

erikap commented on app/templates/main/suppliers.hbs:
```diff
@@ -32,13 +32,24 @@
+              id="new-supplier-id"
+              @value={{this.newSupplierId}}
+              class="mt-1 focus:ring-red-200 focus:border-red-200 block w-full border-gray-300 rounded-md sm:text-sm" />
+          </div>
```
Can you add a 'preview' of the link to RKB such that they can test whether it works with the number they entered

MikiDi replied:
![image](https://github.com/user-attachments/assets/74b04cd8-440c-48a7-af2f-72cdca32634c)

erikap commented on app/components/product/view.hbs:
```diff
@@ -78,10 +78,17 @@
-              <Util::Property @label="Leverancier" @value={{@model.purchaseOffering.businessEntity.name}} />
+              <Util::Property @label="Leverancier">
+                {{@model.purchaseOffering.businessEntity.name}}
+                <Rlv::Button
+                  @skin="link"
+                  @href={{@model.purchaseOffering.businessEntity.rkbLink}}
+                  @icon="external-link-line"
+                />
```
Identifier (and thus link to RKB) may be empty. Best to make showing the icon conditional. Also holds for other places in the interface.

MikiDi committed:
test new supplier link in form

MikiDi committed:
show supplier link conditionally

MikiDi committed:
link out linting

from what I understand, what ember lint suggests here is only applicable
for links out to untrusted environments. rkb is a trusted environment.

MikiDi committed:
focus when opening form

erikap committed:
Remove unused models

erikap committed:
Fix alignment of open and preview links in products list

erikap committed:
Show RKB customer number for supplier if any

erikap committed:
Add supplier number to edit form
